### PR TITLE
Make _get_by_keys return a list instead of a hash reference

### PR DESCRIPTION
**Pull request by mwiencek**
The majority of places that use `_get_by_keys` wrap the result in `values %{}`, so it makes more sense to just return a list.

This was part of my MBS-7241 branch, but since it's taking me a while to finish that, I'm submitting this for review now.

### CODE REVIEW

File: lib/MusicBrainz/Server/Data/Entity.pm [hunk 1]
```diff
@@ -60,11 +48,18 @@ sub get_by_id
     return $result[0];
 }
 
-sub get_by_id_locked
-{
+sub get_by_id_locked {
     my ($self, $id) = @_;
-    my @result = values %{$self->_get_by_keys_append_sql($self->_id_column, 'FOR UPDATE', $id)};
```
Comment (mwiencek) [inline lib/MusicBrainz/Server/Data/Entity.pm]: This was the only thing `_get_by_keys_append_sql` was used for. But the query it's modifying is too simple to be worth factoring this out, I think.

File: lib/MusicBrainz/Server/Data/Entity.pm [hunk 2]
```diff
@@ -60,11 +48,18 @@ sub get_by_id
     return $result[0];
 }
 
-sub get_by_id_locked
-{
+sub get_by_id_locked {
```
Comment (gentlecat) [inline lib/MusicBrainz/Server/Data/Entity.pm]: What does `locked` mean in this case?

**Comment by gentlecat** [comment]
LGTM.

File: lib/MusicBrainz/Server/Data/Entity.pm (hunk 2 above)
Comment (uklauer) [inline lib/MusicBrainz/Server/Data/Entity.pm]: The function not only gets the IDs, but also puts a row lock on them via the `FOR UPDATE` clause. See http://www.postgresql.org/docs/9.1/static/explicit-locking.html#LOCKING-ROWS

Comment (gentlecat) [inline lib/MusicBrainz/Server/Data/Entity.pm]: Huh, that's a nice feature.

**Pull request merged**